internal/monitors/haproxy/metadata.yaml

```diff
@@ -0,0 +1,767 @@
+monitors:
+  - monitorType: haproxy
+    doc: |
+      This monitor scrapes [HAProxy](http://www.haproxy.org/) statistics from an HTTP endpoint or a UNIX socket.
+      It requires HAProxy 1.8+. All metrics have a `process_num` dimension which is the index of the HAProxy
+      process that the metric pertains to.
+
+      <!--- SETUP --->
+      ### HTTP Endpoint Config
+      HAProxy stats must be enabled as described [here](https://www.haproxy.com/blog/exploring-the-haproxy-stats-page/).
+      Define the HAProxy monitor in the agent configuration file and provide the `csv export` url where HAProxy
+      stats are served in csv format. This monitor supports basic HTTP authentication. Simply provide the
+      username and password if required as shown below:
+      ```
+      ...
+      monitors:
+        - type: haproxy
+          url: "http://localhost:8404/stats?stats;csv"
+          username: "my username"
+          password: "my password"
```

Comment:
Should "my username" and "my password" have an underscore or hyphen between the two words? Does HTTP Authentication support spaces?

Reply:
They are just place holders for the actual username and password that the user is going to provide. I changed them to "your username here" and "your password here"